### PR TITLE
Consultant frontpage

### PR DESCRIPTION
**Pull request by IleJok**
Consultant front page now shows active positions with roles matching with preferable roles. Positions are ordered by number of matches in positionSkills and userSkills

### CODE REVIEW

File: front/src/Components/LandingPageMessage.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        <>       
+        <Container>          
+            <h4>Consultant app</h4>
+            <div>Clicking Consultants from the navigation bar takes you to the list of all consultants.</div>
+            <div>Clicking Positions from the navigation bar takes you to the list of all positions.</div>
+            {props.role === "AW" && <div>Clicking Add Position from the navigation bar allows you to submit a new position.</div>}          
+            <div>Clicking My Profile from the navigation bar takes you to your profile.</div>
+        </Container>
+        
+        </>
```
Comment (rrudd) [inline front/src/Components/LandingPageMessage.js]: remove?

File: front/src/Containers/PositionFilter.js [hunk 2]
```diff
@@ -1,66 +1,109 @@
...
+              positionId={position.positionId}
+              description={position.positionDescription}
+              role={position.positionRole}
+              location={position.location}
+              active={position.isActive}
+              company={position.company}
+              status={position.positionStatus}
+              skills={position.positionSkills}
+              favorites={position.userFavorites}
+              createdAt={position.createdAt}
```
Comment (rrudd) [inline front/src/Containers/PositionFilter.js]: maybe pass only position as prop

File: front/src/Containers/PositionSelection.js [hunk 3]
```diff
@@ -161,7 +161,7 @@ class PositionSelection extends Component {
         return this.handleCompare(consultant);
       })
       .sort((a, b) => b.hits - a.hits);
-    return candidates;
+    return candidates;
```
Comment (rrudd) [inline front/src/Containers/PositionSelection.js]: cleanup

File: front/src/Containers/PositionSelection.js [hunk 4]
```diff
@@ -346,7 +346,7 @@ class PositionSelection extends Component {
               )}
           </Col>
         </Row>
-        <Row>{this.renderCandidates()}</Row>
+        {this.state.isEditing === false && <><Row>{this.renderCandidates()}</Row></>}
```
Comment (rrudd) [inline front/src/Containers/PositionSelection.js]: Why the "<>"?

File: front/src/Containers/PositionsList.js [hunk 5]
```diff
@@ -1,47 +1,25 @@
...
-      } else {
-        this.setState({ serverError: response });
-      }
-    });
-  }
-
   renderPositionFilter = () => {
     if (this.state.positions.length > 0) {
-      return <PositionFilter positions={this.state.filteredPositions} />;
+      return <PositionFilter fromNavBar={this.props.fromNavBar} positions={this.props.filteredPositions} onlyActivePositions={this.props.onlyActivePositions}/>;
```
Comment (rrudd) [inline front/src/Containers/PositionsList.js]: remove redundant prop

File: front/src/Containers/PositionsList.js [hunk 6]
```diff
@@ -1,47 +1,25 @@
...
   render() {
     return (
       <>
-        {this.state.serverError !== "" && <Redirect to= {{
-                                          pathname: '/error',
-                                          state: { serverError: this.state.serverError }
-        }} /> }
-        { this.renderPositionFilter() }
+        { this.renderPositionFilter() }        
       </>
```
Comment (rrudd) [inline front/src/Containers/PositionsList.js]: remove redundant "<>"

File: front/src/Containers/Search.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react';
```
Comment (rrudd) [inline front/src/Containers/Search.js]: This component could be called eg. "Positions"

Comment (rrudd) [reply front/src/Containers/Search.js]: or PositionsView

File: front/src/Containers/PositionsList.js (hunk 6 above)
Comment (IleJok) [reply front/src/Containers/PositionsList.js]: doesn't work without them.

**Pull request merged**